metadatastore/commands.py

```diff
@@ -39,88 +89,273 @@ def inner(*args, **kwargs):
+        The RunStart to get the events for.  Can be either
+        a dict or a uid.
 
     Returns
     -------
-    formatted_dict : dict
-        Data key info for the event descriptor that is formatted for the
-        current metadatastore spec.The current metadatastore spec is:
-        {'data_key1':
-         'data_key2': mds.odm_templates.DataKeys
-        }
+    run_stop : doc.Document
+        The RunStop document
     """
-    data_key_dict = {key_name: (
-                     DataKey(**data_key_description) if
-                     not isinstance(data_key_description, DataKey) else
-                     data_key_description)
-                     for key_name, data_key_description
-                     in six.iteritems(data_key_dict)}
-    return data_key_dict
 
+    run_start_uid = doc_or_uid_to_uid(run_start)
+    run_start = runstart_given_uid(run_start_uid)
+    oid = _RUNSTART_UID_to_OID_MAP[run_start['uid']]
 
-def format_events(event_dict):
-    """Helper function for ophyd to format its data dictionary in whatever
-    flavor of the week metadatastore's spec says. This insulates ophyd from
-    changes to the mds spec
+    run_stop = RunStop._get_collection().find_one(
+        {'run_start_id': oid})
 
-    Currently formats the dictionary as {key: [value, timestamp]}
+    if run_stop is None:
+        raise NoRunStop("No run stop exists")
+
+    return _cache_runstop(run_stop)
+
+
+def eventdescriptors_by_runstart(run_start):
+    """Given a RunStart return a list of it's descriptors
+
+    Raises if no EventDescriptors exist.
 
     Parameters
     ----------
-    event_dict : dict
-        The format that ophyd is sending to metadatastore
-        {'data_key1': {
-            'timestamp': timestamp_value, # should be a float value!
-            'value': data_value
-         'data_key2': {...}
-        }
+    run_start : dict or uid
+        The RunStart to get the events for.  Can be either
+        a dict or a uid.
 
     Returns
     -------
-    formatted_dict : dict
-        The event dict formatted according to the current metadatastore spec.
-        The current metadatastore spec is:
-        {'data_key1': [data_value, timestamp_value],
-         'data_key2': [...],
-        }
+    event_descriptors : list
+        A list of EventDescriptor documents
     """
-    return {key: [data_dict['value'], data_dict['timestamp']]
-            for key, data_dict in six.iteritems(event_dict)}
+    # normalize the input and get the runstart oid
+    run_start_uid = doc_or_uid_to_uid(run_start)
+    run_start = runstart_given_uid(run_start_uid)
+    oid = _RUNSTART_UID_to_OID_MAP[run_start['uid']]
+
+    # query the database for any event descriptors which
+    # refer to the given runstart
+    ev_desc_cur = EventDescriptor._get_collection().find(
+        {'run_start_id': oid})
+
+    # loop over the found documents, cache, and dereference
+    rets = [_cache_eventdescriptor(ev_desc) for ev_desc in ev_desc_cur]
```

Comment:
Though I suppose that is dangerous for the case of a run that is currently being collected, since event descriptors can (theoretically) be created at all times during the run and if you have event descriptors in the cache, you will not assume that you need to go check for more.

Though as I type this out, I suppose you could also check to see if there is a RunStop document for the given RunStart and, if so, you do not need to check the db for additional descriptors